File: src/Web/WebApi/src/service/CommentService.js
export default class CommentService {
    constructor(request) {
        this.request = request;
    }

    async create(payload) {
        const response = await this.request({
            method: 'post',
            url: '/comment/create',
            data: payload
        });
        return response?.data ?? response;
    }

    async getList(filter = {}) {
        const response = await this.request({
            method: 'get',
            url: '/comment/list',
            params: filter
        });
        return response?.data ?? response;
    }
}

File: src/Mobile/lib/services/comment_service.dart
import 'package:dio/dio.dart';

/// WebApi `CommentService.js`
class CommentService {
  CommentService(this._dio);

  final Dio _dio;

  Future<dynamic> getList([Map<String, dynamic>? filter]) async {
    final r = await _dio.get<dynamic>('/comment/list', queryParameters: filter);
    return r.data;
  }

  Future<dynamic> create(Map<String, dynamic> payload) async {
    final r = await _dio.post<dynamic>('/comment/create', data: payload);
    return r.data;
  }
}
